spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts' do
+  describe 'GET/index/:id/posts' do
+    it 'should return http success for user`s posts' do
+      get user_posts_url(Post.last)
+      expect(response).to be_successful
+      expect(response).to render_template(:index) # should ensure that the correct template is rendered
+      assert_equal 'index', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is redered
+      assert_match 'Displays a list of posts from a user', @response.body
+    end
+  end
+
+  describe 'GET/index/:id/:id/posts/:id' do
+    it 'should return http success for user`s post' do
+      get user_post_url(user_id: User.last, id: Post.last.id)
+      expect(response).to be_successful
+      expect(response).to render_template(:show) # should ensure that the correct template is rendered
+      assert_equal 'show', @controller.action_name # should ensure the right action is rendered
+      # should correct body for the post view file is redered
+      assert_match 'Shows a single post from a user ...', @response.body
+    end
+  end
```

Comment:
- [ ] Your tests are failing including the show tests in the users file. Kindly fix the issues.

  ![image](https://github.com/ismayito/Rails-blog-app/assets/36542986/ee333df1-c69d-4725-a413-4051d9ad1a53)

Reply:
@TedLivist  could you please ellaborate more because on my machine all tests are passing 
![Screenshot from 2023-12-13 11-23-31](https://github.com/ismayito/Rails-blog-app/assets/47735113/4a79b83b-37be-4e14-9779-29d0495154b7)

Reply:
@ismayito these are the errors I'm getting. The variables for running the tests and for use in the routes should be defined along with the tests so that anyone can run them. You have something like `User.last` but in your tests there is nowhere this user is defined. Did you populate the test db on your PC, 'cause I ran the tests again and they're still failing?

![image](https://github.com/ismayito/Rails-blog-app/assets/36542986/1a02e17d-bee0-438c-be67-774ac5992f43)